Rubberduck.VBEEditor/Events/VBENativeServices.cs

```diff
@@ -165,18 +166,16 @@ private static void OnSelectionChanged(IntPtr hwnd)
             if (pane != null) SelectionChanged?.Invoke(_vbe, new SelectionChangedEventArgs(pane));
         }
 
-        private static string _currentLine;
-        public static event EventHandler<AutoCompleteEventArgs> CaretHidden; // not CodeChanged because wouldn't fire on paste
-        private static void OnCaretHidden(IntPtr hwnd)
+        public static event EventHandler<AutoCompleteEventArgs> KeyDown; 
+        private static void OnKeyDown(KeyPressEventArgs e)
         {
-            var pane = GetCodePaneFromHwnd(hwnd);
-            if (pane?.Selection.IsSingleCharacter ?? false)
+            using (var pane = GetCodePaneFromHwnd(e.Hwnd))
             {
-                var args = new AutoCompleteEventArgs(pane);
-                if (_currentLine != args.OldCode && !string.IsNullOrEmpty(args.OldCode))
+                using (var module = pane.CodeModule)
```

Comment:
this here is the owner of the Pane. Don't go run a `using` in the AutoCompletes themselves

Reply:
It's not, it's always a new wrapper.